APIHook.cpp

```diff
@@ -13,47 +13,736 @@
+	{
+		return m_originalDialog->SetFilter(pFilter);
+	}
+
+	/* [local] */ HRESULT STDMETHODCALLTYPE Show(
+	    /* [annotation][unique][in] */ _In_opt_ HWND hwndOwner)
+	{
+		//�Ăяo����Ƃ�m�F�A�e�̐e��NULL�������Chronos�̐ݒ��ʂ���
+		if (!hwndOwner)
+		{
+			return m_originalDialog->Show(hwndOwner);
+		}
+		HWND hWindowOwner = GetParent(hwndOwner);
+		HWND hWindowParent = {0};
+		if (hWindowOwner)
+		{
+			hWindowParent = GetParent(hWindowOwner);
+		}
+		//hWindowParent��NULL�̏ꍇ�́A���̂܂�
+		if (!hWindowParent)
+		{
+			return m_originalDialog->Show(hwndOwner);
+		}
+
+		if (theApp.m_AppSettings.IsEnableUploadRestriction())
+		{
+			CString strMsg(L"�t�@�C�� �A�b�v���[�h�́A�V�X�e���Ǘ��҂ɂ�萧�����Ă��܂��B");
+			if (hWindowParent)
+			{
+				theApp.SB_MessageBox(hWindowParent, strMsg, NULL, MB_OK | MB_ICONWARNING, TRUE);
+			}
+			else
+			{
+				::MessageBoxW(hwndOwner, strMsg, theApp.m_strThisAppName, MB_OK | MB_ICONWARNING);
+			}
+			return E_ACCESSDENIED;
+		}
+		
+		for (;;)
+		{
+			HRESULT hresult = m_originalDialog->Show(hwndOwner);
+			if (FAILED(hresult))
+			{
+				return hresult;
+			}
+
+			LPWSTR wstrSelPath;
+			IShellItem *psi;
+			hresult = this->GetResult(&psi);
+			
+			if (FAILED(hresult))
+			{
+				return hresult;
+			}
+
+			psi->GetDisplayName(SIGDN_DESKTOPABSOLUTEEDITING, &wstrSelPath);
+
+			CString strSelPath(wstrSelPath);
+			strSelPath.MakeUpper();
+			if (strSelPath.IsEmpty())
+				return hresult;
+
+			if (theApp.IsSGMode())
+			{
+				CString strRoot(m_strRootPath);
+				strRoot.MakeUpper();
+				if (strSelPath.Find(strRoot) != 0)
+				{
+					CString strCaption(theApp.m_strThisAppName);
+					CString strMsg;
+					strMsg.Format(L"�A�b�v���[�h�t�H���_�[[%s]�ȊO����̓A�b�v���[�h�ł��܂���B\n\n�w�肵�Ȃ����Ă��������B\n\n�I��ꂽ�ꏊ[%s]", strRoot, (PCWSTR)strSelPath);
+					::MessageBoxW(hwndOwner, strMsg, strCaption, MB_OK | MB_ICONWARNING);
+					continue;
+				}
+			}
+
+			PathRemoveFileSpec(strSelPath.GetBuffer());
+			strSelPath.ReleaseBuffer();
+			theApp.m_strLastSelectUploadFolderPath = strSelPath;
+
+			if (theApp.m_AppSettings.IsEnableLogging() && theApp.m_AppSettings.IsEnableUploadLogging())
+			{
+				WCHAR* ptrFile = NULL;
+				ptrFile = PathFindFileNameW(strSelPath);
+				CString strFileName(ptrFile ? ptrFile : strSelPath);
+				theApp.SendLoggingMsg(LOG_UPLOAD, strFileName, hwndOwner);
+			}
+			return hresult;
+		}
+		return S_OK;
+	}
+
+	HRESULT STDMETHODCALLTYPE QueryInterface(
+	    /* [in] */ REFIID riid,
+	    /* [iid_is][out] */ _COM_Outptr_ void __RPC_FAR* __RPC_FAR* ppvObject)
+	{
+		return m_originalDialog->QueryInterface(riid, ppvObject);
+	}
+
+	ULONG STDMETHODCALLTYPE AddRef(void)
+	{
+		return ++m_referenceCount;
+	}
+
+	ULONG STDMETHODCALLTYPE Release(void)
+	{
+		ULONG referenceCount = --m_referenceCount;
+		if (referenceCount <= 0)
+		{
+			delete this;
+		}
+		return referenceCount;
+	}
+
+private:
+	CComPtr<IFileOpenDialog> m_originalDialog = nullptr;

```

Comment:
```suggestion
	CComPtr<IFileOpenDialog> m_originalDialog;

```

プリミティブ型では無いので多分初期化は要らない（自分で初期化する）と思います。

Reply:
修正しました。